api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -742,6 +743,50 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     return ResponseEntity.ok(response);
   }
 
+  @Override
+  public ResponseEntity<FileDetail> renameNotebook(String workspace, String workspaceName, NotebookRename rename) {
+    String fromBucket = fireCloudService.getWorkspace(workspace, workspaceName)
+      .getWorkspace()
+      .getBucketName();
+    String origPath = NOTEBOOKS_WORKSPACE_DIRECTORY+ "/" + rename.getName();
+    String newName = rename.getNewName();
+    if (!newName.matches("^.+\\.ipynb")) {
+      newName = newName + ".ipynb";
+    }
+    String newPath = NOTEBOOKS_WORKSPACE_DIRECTORY + "/" + newName;
+    BlobId blobId = BlobId.of(fromBucket, origPath);
+    cloudStorageService.copyBlob(blobId, BlobId.of(fromBucket, newPath));
+    cloudStorageService.deleteBlob(blobId);
+    FileDetail fileDetail = new FileDetail();
+    fileDetail.setName(newName);
+    fileDetail.setPath("gs://" + fromBucket + "/" + NOTEBOOKS_WORKSPACE_DIRECTORY + "/" + newName);
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+    fileDetail.setLastModifiedTime(now.getTime());
+    return ResponseEntity.ok(fileDetail);
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> cloneNotebook(String workspace, String workspaceName, String notebookName) {
+    String fromBucket = fireCloudService.getWorkspace(workspace, workspaceName)
+      .getWorkspace()
+      .getBucketName();
+    String origPath = NOTEBOOKS_WORKSPACE_DIRECTORY + "/" + notebookName;
+    String newPath = NOTEBOOKS_WORKSPACE_DIRECTORY + "/" + notebookName.replaceAll("\\.ipynb", " ") + "Clone.ipynb";
```

Comment:
For workspaces, we name them "Clone of X" by default. May make sense to use the same convention here.

Reply:
Mohs and I had a conversation about this... We thought in the context of Cloning the notebooks in this list, that the user may want the clone to appear next to the cloned, which is achieved by Name Clone. If you really think we should use the other convention I will change it.

Reply:
OK, sounds fine for now if we're sorting on notebook name.

Separately, I don't know whether that was actually the intended product behavior or that just happened to be how it was implemented (the mocks are ordered arbitrarily). I could imagine we may want to sort this page (by default) by last access / modification time, in which case the name matters less.